src/react/__tests__/provider.test.js

```diff
@@ -27,6 +27,20 @@ describe('DiProvider', () => {
     });
   });
 
+  it('should expose stable context', () => {
+    const children = jest.fn();
+    const App = () => (
+      <DiProvider use={[]}>
+        <Context.Consumer>{children}</Context.Consumer>
+      </DiProvider>
+    );
+
+    const { rerender } = render(<App />);
+    rerender(<App />);
+
+    expect(children.mock.calls[0][0]).toBe(children.mock.calls[1][0]);
```

Comment:
an interesting test, look like there is a story behind it.

Reply:
This is because context provider memo can be fiddly 😅